app/lib/contexts/GraphContext.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { createContext, useState } from 'react';
+
+export interface GraphContextProps {
+    range: number[]
+    changeRange: (range: number[]) => void;
+}
+
+export const GraphContext = createContext<GraphContextProps>({
+    range: [],
+    changeRange: () => { },
+});
+
+// eslint-disable-next-line no-undef
+let timer: NodeJS.Timeout;
+
+const debounce = (callback: () => any, timeout: number) => () => {
+    clearTimeout(timer);
+    timer = setTimeout(() => {
+        callback();
+    }, timeout);
+};
```

Comment:
Didn't we have a debounce method somewhere else too? We could try extracting this to some util file